# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1521841678 25200
#      Fri Mar 23 14:47:58 2018 -0700
# Branch stable
# Node ID 4a6284b539337ef3428292c538a3e6127e38447f
# Parent  f80e68529bbfdd5f35194dde79cd562b18100109
test: allow more eigenvalues on eigs tests to pass on i386 (bug #53433)

* eigs.m: Allow more eigenvalues as necessary on BIST tests to allow tests to
pass on i386.

diff --git a/scripts/sparse/eigs.m b/scripts/sparse/eigs.m
--- a/scripts/sparse/eigs.m
+++ b/scripts/sparse/eigs.m
@@ -1420,26 +1420,26 @@ endfunction
 %! warning ("off", "Octave:eigs:UnconvergedEigenvalues", "local");
 %! d = eigs (Afun, 100, 4, "lm", opts);
 %! assert (d(3:4), [NaN+1i*NaN; NaN+1i*NaN]);
 %!testif HAVE_ARPACK
 %! A = 1i * magic (100);
 %! opts.v0 = (1:100)';
 %! opts.maxit = 1;
 %! warning ("off", "Octave:eigs:UnconvergedEigenvalues", "local");
-%! d = eigs (A, 4, "lm", opts);
-%! assert (d(4), NaN+1i*NaN);
+%! d = eigs (A, 5, "lm", opts);
+%! assert (d(5), NaN+1i*NaN);
 %!testif HAVE_ARPACK
 %! A = 1i * magic (100) + eye (100);
 %! opts.v0 = (1:100)';
 %! opts.maxit = 7;
 %! warning ("off", "Octave:eigs:UnconvergedEigenvalues", "local");
 %! d = eigs (A, 10, "sm", opts);
 %! assert (d(9:10), [NaN+1i*NaN; NaN+1i*NaN]);
 %!testif HAVE_ARPACK
 %! A = 1i * magic (100);
 %! Afun = @(x) A * x;
 %! opts.v0 = (1:100)';
 %! opts.maxit = 1;
 %! opts.isreal = false;
 %! warning ("off", "Octave:eigs:UnconvergedEigenvalues", "local");
-%! d = eigs (Afun, 100, 4, "lm", opts);
-%! assert (d(4), NaN+1i*NaN);
+%! d = eigs (Afun, 100, 6, "lm", opts);
+%! assert (d(6), NaN+1i*NaN);
